### PR TITLE
Announcement Cog

### PR DESCRIPTION
**Pull request by Kitty-Therapist**
Finally got the announcement cog to work for this bot. 

It should allow you to grab a channel, and then uses the said channel to send an announcement as long that the bot has permissions.

I've tested it, and it seems to be working effectively.

### CODE REVIEW

File: ServerTools.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import asyncio
+import discord
+from discord.ext import commands
+from discord.ext.commands import BadArgument, Greedy, MemberConverter, RoleConverter
```
Comment (Tarek1337) [inline ServerTools.py]: Get rid of this line, you are not using any of these things.

File: ServerTools.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import asyncio
+import discord
+from discord.ext import commands
+from discord.ext.commands import BadArgument, Greedy, MemberConverter, RoleConverter
+
+import GearBot
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, Configuration, Utils, GearbotLogging, Pages, Emoji, Translator, \
+   Confirmation, MessageUtils
+from Util.Converters import DiscordUser, PotentialID, RoleMode, Guild, \
```
Comment (Tarek1337) [inline ServerTools.py]: Thse are nto used either

File: ServerTools.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import asyncio
+import discord
+from discord.ext import commands
+from discord.ext.commands import BadArgument, Greedy, MemberConverter, RoleConverter
+
+import GearBot
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, Configuration, Utils, GearbotLogging, Pages, Emoji, Translator, \
```
Comment (Tarek1337) [inline ServerTools.py]: Only a few things are used here, remove the unused ones.

**Comment by Kitty-Therapist** [comment]
Thanks. I've removed the unnecessary lines and adjusted them. As for the error, I was told by AEnterprise to make the string for the error where the bot didn't have SEND_MESSAGE permission. 

Lemme know if I missed a line that shouldn't be needed.

File: ServerTools.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+class ServerTools:
+    permissions = {
+        "min": 3,
+        "max": 6,
+        "required": 3
+    }
+    def __init__(self, bot):
+        self.bot: GearBot = bot
+        self.running = True
```
Comment (AEnterprise) [inline ServerTools.py]: why are you keeping track on if it's running or a set for handling stuff this cog doesn't do any of that?

File: lang/en_US.json [hunk 5]
```diff
@@ -547,5 +547,6 @@
   "emoji_upload_help": "Uploads a new emoji",
   "emoji_upload_no_attachments": "Your message did not have any attachments to be uploaded as emoji",
   "emoji_name_space": "Emoji names can not have spaces",
-  "emoji_upload_rejected": "Discord refused the emoji, maybe it has a bad name or is corrupt?"
+  "emoji_upload_rejected": "Discord refused the emoji, maybe it has a bad name or is corrupt?",
+  "announce_invaild_id": "I was not able to talk in that channel. Are you sure that you have the correct channel and I have permission to view and talk in that particular channel?"
```
Comment (AEnterprise) [inline lang/en_US.json]: a bot can always 'see' a channel, if if you put read messages on explicitly denied, it still knows the channel is there, it just denies sending messages there

File: ServerTools.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    async def __local_check(self, ctx):
+        return Permissioncheckers.check_permission(ctx)
+
+    @commands.guild_only()       
+    @commands.command()
+    async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
+        if message != None:
+            try:
+                await channel.send(f"{message}")
```
Comment (AEnterprise) [inline ServerTools.py]: little weird to put fstring here if you only use the var and nothing else in it?

File: ServerTools.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        return Permissioncheckers.check_permission(ctx)
+
+    @commands.guild_only()       
+    @commands.command()
+    async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
+        if message != None:
+            try:
+                await channel.send(f"{message}") 
+            except discord.Forbidden:
+                await ctx.send(
```
Comment (AEnterprise) [inline ServerTools.py]: can you use `MessageUtils.send_to(ctx, 'NO', 'announce_invaild_id')` instead? cleaner and i'm slowly working on routing all these types of messages through that central function

File: ServerTools.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        "max": 6,
+        "required": 3
+    }
+    def __init__(self, bot):
+        self.bot: GearBot = bot
+        self.running = True
+        self.handling = set()
+
+    def __unload(self):
+        self.running = False
```
Comment (AEnterprise) [inline ServerTools.py]: same as above, we're not using any background tasks, can just be removed

**Comment by Kitty-Therapist** [comment]
Made the changes as per request. 👍

File: ServerTools.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+    async def __local_check(self, ctx):
+        return Permissioncheckers.check_permission(ctx)
+
+    @commands.guild_only()       
+    @commands.command()
+    async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
+        if message != None:
+            try:
+                await channel.send("{message}")
```
Comment (Tarek1337) [inline ServerTools.py]: This will just send ``{message}`` to Discord, I don't think you want that.

File: ServerTools.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import asyncio
+import discord
+from discord.ext import commands
+
+import GearBot
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, GearbotLogging, Emoji, Translator, MessageUtils
```
Comment (Tarek1337) [inline ServerTools.py]: You still import stuff not used.

File: ServerTools.py [hunk 11]
```diff
@@ -28,9 +25,9 @@ def __unload(self):
     async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
         if message != None:
             try:
-                await channel.send(f"{message}") 
+                await channel.send("{message}") 
             except discord.Forbidden:
-                await ctx.send(
-                        f"{Emoji.get_chat_emoji('NO')} {Translator.translate('announce_invaild_id', ctx)}")
+                MessageUtils.send_to(ctx, 'NO', 'announce_invaild_id')
+
```
Comment (AEnterprise) [inline ServerTools.py]: this needs to be awaited

File: ServerTools.py [hunk 12]
```diff
@@ -14,11 +14,8 @@ class ServerTools:
     }
     def __init__(self, bot):
         self.bot: GearBot = bot
-        self.running = True
         self.handling = set()
```
Comment (AEnterprise) [inline ServerTools.py]: this line also isn't needed

**Comment by AEnterprise** [comment]
closing this due to age and that i have plans to do this more extensive on the dashboard instead

**Pull request closed without merging**